common/data_refinery_common/models/models.py

```diff
@@ -407,11 +412,24 @@ def __str__(self):
     # This matters, for instance salmon 0.9.0 indexes don't work with 0.10.0
     salmon_version = models.CharField(max_length=255)
 
+    # S3 Key
+    s3_key = models.CharField(max_length=255, default="")
+
     # Common Properties
     is_public = models.BooleanField(default=True)
     created_at = models.DateTimeField(editable=False, default=timezone.now)
     last_modified = models.DateTimeField(default=timezone.now)
 
+    def upload_to_s3(self, absolute_file_path, bucket_name, logger):
+        if bucket_name != "NO_BUCKET":
+            s3_key = self.organism.name + '_' + self.index_type + '.tar.gz'
+            S3.upload_file(absolute_file_path, bucket_name, s3_key,
```

Comment:
If possible, we should `util`-ize all of the S3 logic so that it can be repeated for all of the different objects which need to be stored and retrieved from S3.

If you need a reference: https://github.com/Miserlou/Zappa/blob/192948eef9cfaaa58cd683c0c9a9afa0ecb31fb4/zappa/core.py#L902 has a bit more sanding on it for weird regional quirks.

Reply:
This will also help with testability.